classes/check/dirsizes.php

```diff
@@ -61,19 +61,83 @@ public function get_result(): result {
         return new result(result::INFO, $summary, $details);
     }
     /**
-     * Get a paths sizet
-     * @param string $cfg the path to check
-     * @return string size for a path as html
+     * Get a path's size
+     *
+     * @param  string $cfg the path to check
+     * @return string $size for a path as html
      */
     private function dirsize(string $cfg) {
         global $CFG;
         if (!property_exists($CFG, $cfg)) {
             return "<br>\$CFG->$cfg not in use";
         }
-        $path = $CFG->{$cfg};
-        $size = get_directory_size($path);
+
+        // If Totara, use Totara-compatible function.
+        if (!empty($CFG->totara_version)) {
+            $size = $this->dirsize_totara($cfg);
+        } else {
+            $path = $CFG->{$cfg};
+            $size = get_directory_size($path);
+        }
 
         return "<br>\$CFG->{$cfg} = " . display_size($size);
     }
 
+    /**
+     * Get a path's size (compatible with Totara)
+     *
+     * @param  string $cfg the path to check
+     * @return int $size size for a path as an integer
+     */
+    private function dirsize_totara(string $cfg): int {
+        global $CFG;
+        if (!property_exists($CFG, $cfg)) {
+            return 0;
+        }
+        $rootdir = $CFG->{$cfg};
+
+        return $this->dirsize_totara_helper($rootdir);
+    }
+
+    /**
+     * Recursively calculate the size of a directory (Totara-compatible).
+     *
+     * This replicates Moodle core's get_directory_size() logic,
+     * but avoids using it directly for Totara compatibility.
+     *
+     * @param  string $dir The directory path to measure
+     * @return int Total size in bytes
+     */
+    private function dirsize_totara_helper(string $dir): int {
```

Comment:
How about to rename it to match what the method is actually doing? That would be much easier to understand just reading method's name. 

E.g. `get_directory_size_totara` ?